### PR TITLE
Improve display of timeline date range

### PR DESCRIPTION
**Pull request by dregad**
Before:
selection_001

After:
screenshot-20170124-1149

Fixes [#22131](http://www.mantisbt.org/bugs/view.php?id=22131)

For the record, below is the initial proposal (prior to implementation of suggested changes)
selection_002

### CODE REVIEW

File: core/timeline_inc.php [hunk 1]
```diff
@@ -34,41 +34,41 @@
 
 
 <div id="timeline" class="widget-box widget-color-blue2 <?php echo $t_block_css ?>">
-    <div class="widget-header widget-header-small">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 2]
```diff
@@ -34,41 +34,41 @@
...
-        <h4 class="widget-title lighter">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 3]
```diff
@@ -34,41 +34,41 @@
...
-            <i class="ace-icon fa fa-clock-o"></i>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 4]
```diff
@@ -34,41 +34,41 @@
...
-            <?php echo lang_get( 'timeline_title' ) ?>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 5]
```diff
@@ -34,41 +34,41 @@
...
-        </h4>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 6]
```diff
@@ -34,41 +34,41 @@
...
-        <div class="widget-toolbar">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 7]
```diff
@@ -34,41 +34,41 @@
...
-            <a data-action="collapse" href="#">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 8]
```diff
@@ -34,41 +34,41 @@
...
-                <i class="1 ace-icon fa <?php echo $t_block_icon ?> bigger-125"></i>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 9]
```diff
@@ -34,41 +34,41 @@
...
-            </a>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 10]
```diff
@@ -34,41 +34,41 @@
...
-        </div>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 11]
```diff
@@ -34,41 +34,41 @@
...
-    </div>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 12]
```diff
@@ -34,41 +34,41 @@
...
+	<div class="widget-header widget-header-small">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 13]
```diff
@@ -34,41 +34,41 @@
...
+		<h4 class="widget-title lighter">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 14]
```diff
@@ -34,41 +34,41 @@
...
+			<i class="ace-icon fa fa-clock-o"></i>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 15]
```diff
@@ -34,41 +34,41 @@
...
+			<?php echo lang_get( 'timeline_title' ) ?>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 16]
```diff
@@ -34,41 +34,41 @@
...
+		</h4>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 17]
```diff
@@ -34,41 +34,41 @@
...
+		<div class="widget-toolbar">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 18]
```diff
@@ -34,41 +34,41 @@
...
+			<a data-action="collapse" href="#">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 19]
```diff
@@ -34,41 +34,41 @@
...
+				<i class="1 ace-icon fa <?php echo $t_block_icon ?> bigger-125"></i>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 20]
```diff
@@ -34,41 +34,41 @@
...
+			</a>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 21]
```diff
@@ -34,41 +34,41 @@
...
+		</div>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 22]
```diff
@@ -34,41 +34,41 @@
...
+	</div>
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 23]
```diff
@@ -34,41 +34,41 @@
...
 
-    <div class="widget-body">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 24]
```diff
@@ -34,41 +34,41 @@
...
-        <div class="widget-toolbox">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 25]
```diff
@@ -34,41 +34,41 @@
...
-            <div class="btn-toolbar">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 26]
```diff
@@ -34,41 +34,41 @@
...
-                <?php
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 27]
```diff
@@ -34,41 +34,41 @@
...
-                $t_short_date_format = config_get( 'short_date_format' );
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 28]
```diff
@@ -34,41 +34,41 @@
...
-                echo '&#160;&#160;';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 29]
```diff
@@ -34,41 +34,41 @@
...
-                echo '<span class="label label-yellow"> ' . date( $t_short_date_format, $t_start_time ) . ' </span>';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 30]
```diff
@@ -34,41 +34,41 @@
...
-                echo  ' .. ';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 31]
```diff
@@ -34,41 +34,41 @@
...
-                echo '<span class="label label-yellow"> ' . date( $t_short_date_format, $t_end_time ) . ' </span>';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 32]
```diff
@@ -34,41 +34,41 @@
...
-                echo '&#160;&#160;';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 33]
```diff
@@ -34,41 +34,41 @@
...
+	<div class="widget-body">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 34]
```diff
@@ -34,41 +34,41 @@
...
+		<div class="widget-toolbox">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 35]
```diff
@@ -34,41 +34,41 @@
...
+			<div class="btn-toolbar">
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 36]
```diff
@@ -34,41 +34,41 @@
...
+				<?php
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 37]
```diff
@@ -34,41 +34,41 @@
...
+				$t_short_date_format = config_get( 'short_date_format' );
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 38]
```diff
@@ -34,41 +34,41 @@
...
+				echo '&#160;&#160;';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 39]
```diff
@@ -34,41 +34,41 @@
...
+				echo '<span class="label label-yellow"> ' . date( $t_short_date_format, $t_start_time ) . ' </span>';
+				echo  ' .. ';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 40]
```diff
@@ -34,41 +34,41 @@
...
+				echo '<span class="label label-yellow"> ' . date( $t_short_date_format, $t_end_time ) . ' </span>';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 41]
```diff
@@ -34,41 +34,41 @@
...
+				echo '&#160;&#160;';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 42]
```diff
@@ -34,41 +34,41 @@
...
 
-                echo '<div class="btn-group">';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 43]
```diff
@@ -34,41 +34,41 @@
...
-                echo ' <a class="btn btn-primary btn-xs btn-white btn-round" href="my_view_page.php?days=' .
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 44]
```diff
@@ -34,41 +34,41 @@
...
-                    ( $f_days + 7 ) . '">' . lang_get( 'prev' ) . '</a>';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 45]
```diff
@@ -34,41 +34,41 @@
...
+				echo '<div class="btn-group">';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 46]
```diff
@@ -34,41 +34,41 @@
...
+				echo ' <a class="btn btn-primary btn-xs btn-white btn-round" href="my_view_page.php?days=' .
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 47]
```diff
@@ -34,41 +34,41 @@
...
+					( $f_days + 7 ) . '">' . lang_get( 'prev' ) . '</a>';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 48]
```diff
@@ -34,41 +34,41 @@
...
 
-                $t_next_days = ( $f_days - 7 ) > 0 ? $f_days - 7 : 0;
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 49]
```diff
@@ -34,41 +34,41 @@
...
+				$t_next_days = ( $f_days - 7 ) > 0 ? $f_days - 7 : 0;
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 50]
```diff
@@ -34,41 +34,41 @@
...
 
-                if( $t_next_days != $f_days ) {
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 51]
```diff
@@ -34,41 +34,41 @@
...
-                    echo ' <a class="btn btn-primary btn-xs btn-white btn-round" href="my_view_page.php?days=' .
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 52]
```diff
@@ -34,41 +34,41 @@
...
-                        $t_next_days . '">' . lang_get( 'next' ) . '</a>';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 53]
```diff
@@ -34,41 +34,41 @@
...
-                }
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 54]
```diff
@@ -34,41 +34,41 @@
...
+				if( $t_next_days != $f_days ) {
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 55]
```diff
@@ -34,41 +34,41 @@
...
+					echo ' <a class="btn btn-primary btn-xs btn-white btn-round" href="my_view_page.php?days=' .
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 56]
```diff
@@ -34,41 +34,41 @@
...
+						$t_next_days . '">' . lang_get( 'next' ) . '</a>';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 57]
```diff
@@ -34,41 +34,41 @@
...
+				}
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 58]
```diff
@@ -34,41 +34,41 @@
...
 
-            echo '</div></div></div><div class="widget-main no-padding"><div class="profile-feed">';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 59]
```diff
@@ -34,41 +34,41 @@
...
+			echo '</div></div></div><div class="widget-main no-padding"><div class="profile-feed">';
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 60]
```diff
@@ -82,5 +82,5 @@
 		echo '</div>';
 	} else {
 		timeline_print_events( $t_events );
-    }
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

File: core/timeline_inc.php [hunk 61]
```diff
@@ -82,5 +82,5 @@
...
+	}
```
Comment (k1631485) [inline core/timeline_inc.php]: whitespace change

**Comment by syncguru** [comment]
@dregad 
I think the header is too small to pack date ranges in addition to 'Prev', 'Next' buttons. Have you tested this with other languages? How does it look on mobile or small screens? snapshots?

File: core/timeline_inc.php [hunk 62]
```diff
@@ -37,41 +37,46 @@
...
 		<h4 class="widget-title lighter">
 			<i class="ace-icon fa fa-clock-o"></i>
-			<?php echo lang_get( 'timeline_title' ) ?>
+			<?php
+				echo lang_get( 'timeline_title' );
+
+				# Display date range
+				$t_short_date_format = config_get( 'short_date_format' );
+				echo '<span class="badge"> '
+					. date( $t_short_date_format, $t_start_time )
```
Comment (k1843568) [inline core/timeline_inc.php]: use ,

File: core/timeline_inc.php [hunk 63]
```diff
@@ -37,41 +37,46 @@
...
+					. ' .. '
```
Comment (k1843568) [inline core/timeline_inc.php]: use ,

File: core/timeline_inc.php [hunk 64]
```diff
@@ -37,41 +37,46 @@
...
+					. date( $t_short_date_format, $t_end_time )
```
Comment (k1843568) [inline core/timeline_inc.php]: use ,

File: core/timeline_inc.php [hunk 65]
```diff
@@ -37,41 +37,46 @@
...
+					. ' </span>';
```
Comment (k1843568) [inline core/timeline_inc.php]: use ,

File: core/timeline_inc.php [hunk 66]
```diff
@@ -37,41 +37,46 @@
...
-
-				if( $t_next_days != $f_days ) {
-					echo ' <a class="btn btn-primary btn-xs btn-white btn-round" href="my_view_page.php?days=' .
-						$t_next_days . '">' . lang_get( 'next' ) . '</a>';
-				}
-
-			echo '</div></div></div><div class="widget-main no-padding"><div class="profile-feed">';
-
-	echo '</div></div>';
+					$t_next_days = max( $f_days - 7, 0 );
```
Comment (k1843568) [inline core/timeline_inc.php]: extra t_ in variable name

File: core/timeline_inc.php [hunk 67]
```diff
@@ -37,41 +37,46 @@
...
+					if( $t_next_days != $f_days ) {
```
Comment (k1843568) [inline core/timeline_inc.php]: extra f_ in variable name

File: core/timeline_inc.php [hunk 68]
```diff
@@ -37,41 +37,46 @@
...
+						print_small_button(
+							'my_view_page.php?days=' . $t_next_days,
```
Comment (k1843568) [inline core/timeline_inc.php]: extra t_ in variable name

File: core/timeline_inc.php [hunk 69]
```diff
@@ -37,41 +37,46 @@
...
-				echo  ' .. ';
-				echo '<span class="label label-yellow"> ' . date( $t_short_date_format, $t_end_time ) . ' </span>';
-				echo '&#160;&#160;';
-
-				echo '<div class="btn-group">';
-				echo ' <a class="btn btn-primary btn-xs btn-white btn-round" href="my_view_page.php?days=' .
-					( $f_days + 7 ) . '">' . lang_get( 'prev' ) . '</a>';
+					# Display Prev/Next buttons
+					print_small_button(
+						'my_view_page.php?days=' . ( $f_days + 7 ),
```
Comment (k1843568) [inline core/timeline_inc.php]: extra f_ in variable name

File: core/timeline_inc.php [hunk 70]
```diff
@@ -37,41 +37,46 @@
 	<div class="widget-header widget-header-small">
 		<h4 class="widget-title lighter">
 			<i class="ace-icon fa fa-clock-o"></i>
-			<?php echo lang_get( 'timeline_title' ) ?>
+			<?php
+				echo lang_get( 'timeline_title' );
+
+				# Display date range
+				$t_short_date_format = config_get( 'short_date_format' );
```
Comment (k1843568) [inline core/timeline_inc.php]: extra t_ in variable name

File: core/timeline_inc.php (hunk 62 above)
Comment (k1843568) [inline core/timeline_inc.php]: extra t_ in variable name

**Comment by dregad** [comment]
> Have you tested this with other languages? How does it look on mobile or small screens? snapshots?

Yes. If *timeline_title* string is too long to fit in the browser's width, the dates (and buttons) just wrap to next line, similar to what it does for the other boxes. Below is a sample with Portuguese, with a window as narrow as I can make it

selection_003

It's worth mentioning that with such a narrow display, there are other rendering issues (e.g. the navbar div overlapping the main container area, and thus hiding some elements such as the nav-tabs), but that's a different discussion.

**Comment by syncguru** [comment]
The second image will wrap twice once the second button for range navigation is added. That would make it unusable. I think the range & navigation buttons should go on its own row (as it was) or to the footer of the box.

**Comment by dregad** [comment]
In that case your advice for styling it nicely would be appreciated. I'm not a UI guy.

**Comment by k2152262** [comment]
@vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @dentxinho @bieli @tomcarlson @fengkuok @syncguru @diegovice @caribe @fager @pikiel @nikoutel @6run0 @AgnidykhBayky  @sayyidsh @pixelstreamAu @johnms @mikeching @g-taki @benv8nam @stew721 @sayazyi @mamoniem @lezouretalice @suriyaakudolsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @zedxxx @krime @scottmatheina @faviogalvis @vand3rson @like9729 @badfiles @zenithtekla @sureshinde @peterdavehello @netsf @csatwood @miao606 @bitlin @ioroid @supertassu @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @boby75 @lordshell @burningwind  @nospam1961 @planetbrett @gantha @pbcreativesolutions @bastilimbach @ltpgt @modir  progress on implementing feedback ?

**Comment by avelino** [comment]
Why am I getting notification of this repository? Remove me

**Comment by xasos** [comment]
@k2152262 please stop mentioning large groups of people like this. It's not good for our notifications. If you need an RFC, add the relevant people to the thread.

**Comment by mantis** [comment]
@vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @dentxinho @bieli @tomcarlson @fengkuok @syncguru @diegovice @caribe @fager @pikiel @nikoutel @6run0 @AgnidykhBayky @sayyidsh @pixelstreamAu @johnms @mikeching @g-taki @benv8nam @stew721 @sayazyi @mamoniem @lezouretalice @suriyaakudolsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @zedxxx @krime @scottmatheina @faviogalvis @vand3rson @like9729 @badfiles @zenithtekla @sureshinde @peterdavehello @netsf @csatwood @miao606 @bitlin @ioroid @supertassu @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @boby75 @lordshell @burningwind @nospam1961 @planetbrett @gantha @pbcreativesolutions @bastilimbach @ltpgt @modir @dharmeshkakadia @xasos @LewisJEllis whats status on this PR?

**Comment by mantis** [comment]
@k2152262 why am i getting this ?

**Comment by dharmeshkakadia** [comment]
The account now seems to be 404. IS this some form of github spam or something ?

**Comment by dregad** [comment]
@dharmeshkakadia I think that's what it is indeed. We've been hit 5-6 times by such attacks over the past few days, and they seem to be getting worse (the earlier ones did not @mention anyone so only thread subscribers would be hit). As soon as I notice them I block the user and notify Github support.

**Comment by dregad** [comment]
Ping @syncguru, in case you missed [my last note](https://github.com/mantisbt/mantisbt/pull/985#issuecomment-271311259) due to the spam... Thanks in advance !

**Comment by syncguru** [comment]
@dregad I did not forget, just busy .. hopefully I have something for you in the next few days

**Comment by dregad** [comment]
No worries, cheers !

**Comment by zedxxx** [comment]
@mantis WTF why I'm getting notifications from this repository? Please, remove me from all notification lists!

**Comment by PeterDaveHello** [comment]
@zedxxx I got the same problem, maybe we are just tagged by a spam account, you can unsubscribe this issue from the right side menu, there is a button for it, and it'll show you the reason why you received it, for me, it says I'm mentioned ...

**Comment by zedxxx** [comment]
@PeterDaveHello 

> you can unsubscribe this issue from the right side menu

Yes, I can and I do this since yesterday. I've got hundred mails and they still coming! I hate this f***ing spammer!

**Comment by syncguru** [comment]
@dregad I think the current layout works well with different locale and screen sizes. Since the background color of the date ranges is the issue here, I thought we should start with that:

screenshot from 2017-01-20 18-49-26

**Comment by sync1** [comment]
@vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  whats status ?

**Comment by dregad** [comment]
That would be fine, thanks for the feedback. Would you mind pasting the relevant CSS changes ?

**Comment by syncguru** [comment]
If you inspect the css classes used with the data range, you should see lable-yellow. All you need to do is to change to label-grey. That's it, you should be all set.

**Comment by dregad** [comment]
Pull request updated based on @syncguru's suggestion; also changed target and rebased to _master_ (instead of master-2.0.x as originally submitted)

File: core/timeline_inc.php [hunk 71]
```diff
@@ -67,20 +66,30 @@
...
+			</div>
+		</div>
 
-			echo '</div></div></div><div class="widget-main no-padding"><div class="profile-feed">';
-
-	echo '</div></div>';
+		<div class="widget-main no-padding">
+			<div class="profile-feed">
+			</div>
+		</div>
```
Comment (dregad) [inline core/timeline_inc.php]: @syncguru are these 2 divs really useful ? It looks like they are not doing anything (I don't see any differences in display if I remove them), and are printed later on (by _timeline_print_events()_ call), so I'm wondering if they are maybe a leftover of older code.

**Comment by dregad** [comment]
That's because I fixed indentation with spaces instead of tabs and reformatted code. Please review individual commit af00b8698f6a9b5b7e0849ea481d6a0554d698e1

File: core/timeline_inc.php [hunk 72]
```diff
@@ -32,7 +32,6 @@
 $t_block_icon = $t_collapse_block ? 'fa-chevron-down' : 'fa-chevron-up';
 ?>
 
-
```
Comment (l266509) [inline core/timeline_inc.php]: @vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  this looks like commit contains additional whitespace change

File: core/timeline_inc.php [hunk 73]
```diff
@@ -49,7 +48,7 @@
 	<div class="widget-body">
 		<div class="widget-toolbox">
 			<div class="btn-toolbar">
-				<?php
```
Comment (l266509) [inline core/timeline_inc.php]: @vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  this looks like commit contains additional whitespace change

File: core/timeline_inc.php [hunk 74]
```diff
@@ -67,20 +66,30 @@
...
+				echo '</div>';
+?>
+			</div>
+		</div>
 
-			echo '</div></div></div><div class="widget-main no-padding"><div class="profile-feed">';
-
-	echo '</div></div>';
+		<div class="widget-main no-padding">
+			<div class="profile-feed">
```
Comment (l266509) [inline core/timeline_inc.php]: @vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  <div />

File: core/timeline_inc.php [hunk 75]
```diff
@@ -67,20 +66,30 @@
...
+		echo '<a class="btn btn-primary btn-sm btn-white btn-round" href="my_view_page.php?days='
+			. $f_days . '&amp;all=1">' . lang_get( 'timeline_more' ) . '</a>';
 		echo '</div>';
 		echo '</div>';
 	} else {
 		timeline_print_events( $t_events );
 	}
-echo '</div></div>';
+?>
+
```
Comment (l266509) [inline core/timeline_inc.php]: @vboctor @giallu @daryn @rombert @atrol @grangeway @dregad @Dentxinho @bieli @tomcarlson @syncguru @diegovice @caribe @fager @nikoutel @6RUN0 @AgnidykhBayky @sayyidsh @PixelstreamAU @JohnMS @mikeching @g-taki @benv8nam @stew721 @sayazyi @lezouretalice @SuriyaaKudoIsc @solovets @kacmem @xiaojiangang @chenyongze @myu9882 @markmarco16 @mgretler @lincolnwung @krime @scottmatheina @FavioGalvis @vand3rson @like9729 @badfiles @zenithtekla @SureShinde @netsf @csatwood @miao606 @mdj20130907 @bitlin @ioroid @supertassu @TasCraft @fernandodelucca @ulkuderner @vbathke @hersonpc @jrtorres42 @thejuanwhocode @cyuankai @mantianxingxing @shirleycharlin @boby75 @Lordshell @BurningWind @jaguarxii @Praymundo @johnzastrow @ewmartinez @marcioklein @lypzln @nospam1961 @jllano @lincoln56 @modir @xiaocnie @sivpro @planetbrett @teodortalov @ltpgt @pbcreativesolutions @Gantha @Gerolkae @gunsmoker @nikerabbit @translatewiki  this looks like commit contains additional whitespace change

File: core/timeline_inc.php (hunk 71 above)
Comment (dregad) [inline core/timeline_inc.php]: @syncguru you haven't answered this question. Can these 2 divs be removed ?

**Pull request merged**